src/tiled/editpolygontool.cpp

```diff
@@ -618,6 +664,15 @@ void EditPolygonTool::showHandleContextMenu(PointHandle *clickedHandle,
     connect(joinNodesAction, SIGNAL(triggered()), SLOT(joinNodes()));
     connect(splitSegmentsAction, SIGNAL(triggered()), SLOT(splitSegments()));
 
+    const auto firstHandle = *mSelectedHandles.begin();
+    MapObject *mapObject = firstHandle->mapObjectItem()->mapObject();
+
+    if (mapObject->shape() == MapObject::Polyline) {
+        QAction *extendPolyline = menu.addAction(tr("Extend Polyline"));
+        extendPolyline->setEnabled(n == 1 && firstHandle->pointIndex() == mapObject->polygon().size() - 1);
```

Comment:
I found this boolean expression a bit hard to read... Maybe passing it into a variable could help here (would also shorten the line a bit). But that might be personal taste...